shared/provision/username-or-email/index.js

```diff
@@ -0,0 +1,116 @@
+// @flow
+// TODO remove Container
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+import Container from '../../login/forms/container'
+import * as Constants from '../../constants/provision'
+
+type Props = {|
+  badUsernameError: boolean,
+  error: string,
+  onBack: () => void,
+  onGoToSignup: () => void,
+  onSubmit: (usernameOrEmail: string) => void,
+  submittedUsernameOrEmail: string,
+|}
+
+const BadUsernameError = (props: {|onGoToSignup: () => void|}) => (
+  <Kb.Box2 direction="vertical" centerChildren={true}>
+    <Kb.Text type="BodySmallError" style={styles.error}>
+      This username or email doesn't exist.
+    </Kb.Text>
+    <Kb.Text onClick={props.onGoToSignup} style={styles.errorLink} type="BodySmallPrimaryLink">
+      Sign up for a new account?
+    </Kb.Text>
+  </Kb.Box2>
+)
+
+class UsernameOrEmail extends React.Component<Props, State> {
+  state: State
+
+  constructor(props: Props) {
+    super(props)
+    this.state = {usernameOrEmail: ''}
```

Comment:
don't need this constructor

Reply:
Huh -- how would `this.state` be initialized without it?

Reply:
`state = {usernameOrEmail: ''}` above